tensorflow/java/src/main/native/operation_builder_jni.cc

```diff
@@ -257,3 +257,32 @@ JNIEXPORT void JNICALL Java_org_tensorflow_OperationBuilder_setAttrShape(
   TF_SetAttrShape(d, cname, cvalue.get(), static_cast<int>(num_dims));
   env->ReleaseStringUTFChars(name, cname);
 }
+
+JNIEXPORT void JNICALL Java_org_tensorflow_OperationBuilder_setAttrStringList(
+        JNIEnv* env, jclass object, jlong handle, jstring name, jobjectArray values) {
+ static_assert(sizeof(jbyte) == 1,
+                "Require Java byte to be represented as a single byte");
+  TF_OperationDescription* d = requireHandle(env, handle);
+  if (d == nullptr) return;
+  const char *cname = env->GetStringUTFChars(name, nullptr);
+  int num_strings = env->GetArrayLength(values);
+  size_t *lengths = new size_t[num_strings];
+  jbyte **strings = new jbyte *[num_strings];
```

Comment:
Where are these being deallocated? Is there a missing:

```
delete[] lengths;
delete[] strings;
```
in this function?

Better still, use `std::unique_ptr` as is used in other places in this file to avoid worrying about manual deletes (e.g., see definition of `DEFINE_SET_ATTR_LIST`), so this would change to:

```
std::unique_ptr<size_t[]> lengths(new size_t[num_strings]);
std::unique_ptr<jbyte*[]> strings(new jbyte*[num_string]));
```